File: pokemon.py
## Packages
import pandas as pd
import numpy as np

## Importing Pokedex Data
poke_data = pd.read_csv('Pokedex_Ver_SV2.csv')

#Generator Function

def Generator():
    # Selecting random Pokemon from the Pokedex
    global Random_Pokemon
    Random_Pokemon = poke_data['Name'].sample(n=1)
    # Return the random Pokemon
    return Random_Pokemon

def HP():
    # Display the HP of the Pokemon selected from the generator function
     global HP_Stat
     HP_Stat = poke_data['HP'].loc[poke_data['Name'] == Random_Pokemon.values[0]]
     
     # Return the HP sat
     return HP_Stat

def Attack():
    # Display the attack of the Pokemon selected from the generator function
    global Attack_Stat
    Attack_Stat = poke_data['Attack'].loc[poke_data['Name'] == Random_Pokemon.values[0]]
     
    # Return the attack stat
    return Attack_Stat

def Defense():
    # Display the defense of the Pokemon selected from the generator function
    global Defense_Stat
    Defense_Stat = poke_data['Defense'].loc[poke_data['Name'] == Random_Pokemon.values[0]]
    
    # Return the defense stat
    return Defense_Stat

def SP_Attack():
    # Display the special attack of the Pokemon selected from the generator function
    global SP_Attack_Stat
    SP_Attack_Stat = poke_data['SP_Attack'].loc[poke_data['Name'] == Random_Pokemon.values[0]]
    
     # Return the special attack stat
    return SP_Attack_Stat

def SP_Defense():
    # Display the special defense of the Pokemon selected from the generator function
    global SP_Defense_Stat
    SP_Defense_Stat = poke_data['SP_Defense'].loc[poke_data['Name'] == Random_Pokemon.values[0]]
    
    # Return the special defense stat
    return SP_Defense_Stat

def Speed():
    # Display the speed of the Pokemon selected from the generator function
    global Speed_Stat
    Speed_Stat = poke_data['Speed'].loc[poke_data['Name'] == Random_Pokemon.values[0]]
    
    # Return the speed stat
    return Speed_Stat

def Total():
    # Display the sum of the chosen stats from the stat function
    global Total_Stat
    Total_Stat = [HP_Stat.values[0] + Attack_Stat.values[0] + Defense_Stat.values[0] + SP_Attack_Stat.values[0] + SP_Defense_Stat.values[0] + Speed_Stat.values[0]]
    
     # Return the total stat
    return Total_Stat


Generator()

HP()
Attack()
Defense()
SP_Attack()
SP_Defense()
Speed()

Total()

File: pokemon.js
// Assuming you have loaded 'Pokedex_Ver_SV2.csv' into a variable named 'poke_data' in the same format as the pandas DataFrame

// Generator Function
function Generator() {
    // Selecting random Pokemon from the Pokedex
    let Random_Pokemon = poke_data['Name'][Math.floor(Math.random() * poke_data['Name'].length)];

    // Return the random Pokemon
    return Random_Pokemon;
}

function HP() {
    // Display the HP of the Pokemon selected from the generator function
    let HP_Stat = poke_data['HP'].findIndex(row => row['Name'] === Random_Pokemon);
    
    // Return the HP stat
    return HP_Stat;
}

function Attack() {
    // Display the attack of the Pokemon selected from the generator function
    let Attack_Stat = poke_data['Attack'].findIndex(row => row['Name'] === Random_Pokemon);
    
    // Return the attack stat
    return Attack_Stat;
}

function Defense() {
    // Display the defense of the Pokemon selected from the generator function
    let Defense_Stat = poke_data['Defense'].findIndex(row => row['Name'] === Random_Pokemon);
    
    // Return the defense stat
    return Defense_Stat;
}

function SP_Attack() {
    // Display the special attack of the Pokemon selected from the generator function
    let SP_Attack_Stat = poke_data['SP_Attack'].findIndex(row => row['Name'] === Random_Pokemon);
    
    // Return the special attack stat
    return SP_Attack_Stat;
}

function SP_Defense() {
    // Display the special defense of the Pokemon selected from the generator function
    let SP_Defense_Stat = poke_data['SP_Defense'].findIndex(row => row['Name'] === Random_Pokemon);
    
    // Return the special defense stat
    return SP_Defense_Stat;
}

function Speed() {
    // Display the speed of the Pokemon selected from the generator function
    let Speed_Stat = poke_data['Speed'].findIndex(row => row['Name'] === Random_Pokemon);
    
    // Return the speed stat
    return Speed_Stat;
}

function Total() {
    // Display the sum of the chosen stats from the stat function
    let HP_Stat = HP();
    let Attack_Stat = Attack();
    let Defense_Stat = Defense();
    let SP_Attack_Stat = SP_Attack();
    let SP_Defense_Stat = SP_Defense();
    let Speed_Stat = Speed();
    let Total_Stat = poke_data['HP'][HP_Stat] + poke_data['Attack'][Attack_Stat] + poke_data['Defense'][Defense_Stat] + poke_data['SP_Attack'][SP_Attack_Stat] + poke_data['SP_Defense'][SP_Defense_Stat] + poke_data['Speed'][Speed_Stat];
    
    // Return the total stat
    return Total_Stat;
}

function Reset() {
    // Reset all stored variables
    Random_Pokemon = null;
    Attack_Stat = null;
    Defense_Stat = null;
    HP_Stat = null;
    SP_Attack_Stat = null;
    SP_Defense_Stat = null;
    Speed_Stat = null;
    Total_Stat = null;
}
